### DISCUSSION TITLE
Primary Storage Deletion ,  after Zone is deleted.

### DISCUSSION THREAD

**Discussion by jack99trade** [apache/cloudstack#9537] [Q&A] [answered]
Hi all,  I have deleted a Zone and now Primary storage is in disabled state.  

how do perform delete of this Primary storage. can't find the delete option.

**Comment by DaanHoogland** [comment, marked as answer]
@jack99trade I don't think you should be allowed to delete a zone when it still contains any resources, so I wonder how the primary storage could still be there, but for dleting a primary storage you should first put it in maintance. After it is in maintenance deleting it will be allowed.

**Reply by jack99trade** [reply]
Hi @DaanHoogland  ,  Its was an NFS mounted Primary Share and there are only 3 options 
![image](https://github.com/user-attachments/assets/023dd10e-8841-40d3-8988-e61ddbe1bef0)

No option to put in maintenance or delete.

**Reply by jack99trade** [reply]
Its solved now.  Thanks.   I first enabled the Storage then the option for maintenance mode was visible and followed by delete. 
Thanks @DaanHoogland
